packages/block-editor/src/components/block-tools/use-show-block-tools.js

```diff
@@ -29,10 +29,11 @@ export function useShowBlockTools() {
 		const clientId =
 			getSelectedBlockClientId() || getFirstMultiSelectedBlockClientId();
 
-		const block = getBlock( clientId ) || { name: '', attributes: {} };
+		const block = getBlock( clientId );
 		const editorMode = __unstableGetEditorMode();
-		const hasSelectedBlock = clientId && block?.name;
-		const isEmptyDefaultBlock = isUnmodifiedDefaultBlock( block );
+		const hasSelectedBlock = !! clientId && !! block;
+		const isEmptyDefaultBlock =
+			hasSelectedBlock && isUnmodifiedDefaultBlock( block );
```

Comment:
I don't think we should tie the `hasSelectedBlock` condition here to the `isEmptyDefaultBlock`.

Reply:
Why even call `isUnmodifiedDefaultBlock` if there is no selected block? I.e. without a selection it can’t be a empty default block right?